docs/development/submitting-patches.rst

```diff
@@ -57,6 +57,112 @@ Important information should be documented with prose in the ``docs`` section.
 To ensure it builds and passes `doc8`_ style checks you can run
 ``make docs`` and ``make lint`` respectively.
 
+Keeping your local branch updated
+---------------------------------
+
+As you work, you will need to keep your local ``master`` branch up-to-date with
+the ``master`` branch in the main Warehouse repository, which moves forward as
+the maintainers merge pull requests. Most people working on the project use
+this workflow.
+
+The following documentation assumes that you have git configured so that when
+you run the following command:
+
+.. code-block:: console
+
+  git remote -v
+
+Your output looks like this:
+
+.. code-block:: console
+
+  origin  https://github.com/USERNAME/warehouse.git (fetch)
+  origin  https://github.com/USERNAME/warehouse.git (push)
+  upstream  https://github.com/pypa/warehouse.git (fetch)
+  upstream  https://github.com/pypa/warehouse.git (push)
+
+
+In the example above, ``USERNAME`` is your username on GitHub.
+
+First, fetch the latest changes from the main Warehouse repository,
+``upstream``:
+
+.. code-block:: console
+
+  git fetch upstream
+
+Then, check out your local ``master`` branch, and rebase the changes on top of
+it:
+
+.. code-block:: console
+
+  git checkout master
+  git rebase upstream/master
+
+Next, push the updates you have just made to your local ``master`` branch to
+your ``origin`` repository on GitHub:
+
+.. code-block:: console
+
+  git checkout master
+  git push origin master
+
+Now your local ``master`` branch and the ``master`` branch in your ``origin``
+repo have been updated with the most recent changes from the main Warehouse
+repository.
+
+To keep your feature branches updated, the process is similar:
+
+.. code-block:: console
+
+   git checkout awesome-feature
+   git fetch upstream
+   git rebase upstream/master
+
+Now your feature branch has been updated with the latest changes from the
+``master`` branch on the upstream Warehouse repository.
+
+It's good practice to back up your feature branches by pushing them to your
+``origin`` on GitHub as you are working on them. To push a feature branch,
+run this command:
+
+.. code-block:: console
+
+    git push origin awesome-feature
+
+In this example, ``awesome-feature`` is the name of your feature branch. This
+will push the feature branch you are working on to GitHub, but will not
+create a PR.
+
+Once you have pushed your feature branch to your ``origin``, if you need to
+update it again, you will have to force push your changes by running the
+following command:
+
+.. code-block:: console
+
+    git push -f origin awesome-feature
+
+The ``-f`` flag after ``push`` updates your ``origin`` feature branch with the
+changes you have made on your local feature branch.
```

Comment:
You should explain here that `-f` means "force" -- you are implying that it means "feature"!

You should also explain here that if you DO have a pull request open, updating the branch on `origin` will update the PR.